frappe/custom/doctype/property_setter/property_setter.py

```diff
@@ -88,7 +88,7 @@ def make_property_setter(doctype, fieldname, property, value, property_type, for
 	})
 	property_setter.flags.ignore_permissions = True
 	property_setter.flags.validate_fields_for_doctype = validate_fields_for_doctype
-	property_setter.insert()
+	property_setter.insert(db_auto_commit = frappe.flags.in_test or frappe.flags.in_setup or frappe.flags.in_install)
```

Comment:
I'd suggest we find a previous step in the setup wizard or install process which can be described as a single step and keep it atomic instead.

Or increase the `MAX_WRITES_PER_TRANSACTION` for the DB connection in the respective setup or install instance.

Reply:
If you can somehow get this to work upstream in the ERPNext PR I am all ears, I tried several different things before resulting to this. The problem is primarily in india_setup.py for this specific instance and basically causes a bunch of tests to fail for no hsn codes because of a name not commited.

Reply:
Have you tried increasing `frappe.db.MAX_WRITES_PER_TRANSACTION` (or setting it to `math.inf`) and committing post execution for the context of `make_property_setters`? (in https://github.com/frappe/erpnext/blob/3607f51610f4700acb1fd763dee8e2a0082a4a37/erpnext/regional/india/setup.py)

Reply:
I will try this today and report back

Reply:
@gavindsouza I did as you recommended and it appeared to work:

change: https://github.com/frappe/erpnext/pull/28359/commits/d0a6b9b3b163da878198de26ef62594901b9038c
clean maria: https://github.com/frappe/erpnext/runs/4653436143?check_suite_focus=true
clean postgres (minus one flaky unrelated test): https://github.com/frappe/erpnext/runs/4653436083?check_suite_focus=true


I had also called this autocommit in a few other places throughout my Erpnext changes so I am going to back out of those as well and assuming there are no further issues I will go ahead and revert all of these auto_commit changes

Reply:
This is now reverted back to its original state - no more autocommit